starboard/xb1/shared/accessibility_get_caption_settings.cc

```diff
@@ -0,0 +1,194 @@
+    case ClosedCaptionOpacity::OneHundredPercent:
+      return kSbAccessibilityCaptionOpacityPercentage100;
+    case ClosedCaptionOpacity::SeventyFivePercent:
+      return kSbAccessibilityCaptionOpacityPercentage75;
+    case ClosedCaptionOpacity::TwentyFivePercent:
+      return kSbAccessibilityCaptionOpacityPercentage25;
+    case ClosedCaptionOpacity::ZeroPercent:
+      return kSbAccessibilityCaptionOpacityPercentage0;
+  }
+}
+
+}  // namespace.
+
+bool GetCaptionSettings(SbAccessibilityCaptionSettings* caption_settings) {
+  if (!caption_settings ||
+      !starboard::common::MemoryIsZero(
+          caption_settings, sizeof(SbAccessibilityCaptionSettings))) {
+    return false;
+  }
+
+  caption_settings->background_color =
+      ClosedCaptionColorToSbAccessibilityCaptionColor(
+          ClosedCaptionProperties::BackgroundColor);
+  caption_settings->background_color_state = kSbAccessibilityCaptionStateSet;
+
+  caption_settings->background_opacity =
+      ClosedCaptionOpacityToSbAccessibilityCaptionOpacityPercentage(
+          ClosedCaptionProperties::BackgroundOpacity);
+  caption_settings->background_opacity_state = kSbAccessibilityCaptionStateSet;
+
+  caption_settings->character_edge_style =
+      ClosedCaptionEdgeEffectToSbAccessibilityCaptionCharacterEdgeStyle(
+          ClosedCaptionProperties::FontEffect);
+  caption_settings->character_edge_style_state =
+      kSbAccessibilityCaptionStateSet;
+
+  caption_settings->font_color =
+      ClosedCaptionColorToSbAccessibilityCaptionColor(
+          ClosedCaptionProperties::FontColor);
+  caption_settings->font_color_state = kSbAccessibilityCaptionStateSet;
+
+  caption_settings->font_family =
+      ClosedCaptionStyleToSbAccessibilityCaptionFontFamily(
+          ClosedCaptionProperties::FontStyle);
+  caption_settings->font_family_state = kSbAccessibilityCaptionStateSet;
+
+  caption_settings->font_opacity =
+      ClosedCaptionOpacityToSbAccessibilityCaptionOpacityPercentage(
+          ClosedCaptionProperties::FontOpacity);
+  caption_settings->font_opacity_state = kSbAccessibilityCaptionStateSet;
+
+  caption_settings->font_size =
+      ClosedCaptionSizeToSbAccessibilityCaptionFontSizePercentage(
+          ClosedCaptionProperties::FontSize);
+  caption_settings->font_size_state = kSbAccessibilityCaptionStateSet;
+
+  caption_settings->window_color =
+      ClosedCaptionColorToSbAccessibilityCaptionColor(
+          ClosedCaptionProperties::RegionColor);
+  caption_settings->window_color_state = kSbAccessibilityCaptionStateSet;
+
+  caption_settings->window_opacity =
+      ClosedCaptionOpacityToSbAccessibilityCaptionOpacityPercentage(
+          ClosedCaptionProperties::RegionOpacity);
+  caption_settings->window_opacity_state = kSbAccessibilityCaptionStateSet;
+
+  caption_settings->is_enabled = true; // TODO: check it
+  caption_settings->supports_is_enabled = true;
+  caption_settings->supports_set_enabled = false;
+  caption_settings->supports_override = true;
```

Comment:
ClosedCaptionProperties doesn't provide info like that

Reply:
so, the question is how should I detect if caption_settings is enabled or not?

Reply:
I took at look at the UWP page and there's no obvious documentation for where it would be labelled as enabled or disabled. I even poked around some other media property settings and couldn't find something.

I would check a couple of things, like what do each of the property calls return if you have it enabled on your console vs disabled on your console. Maybe they simply set both font and region opacity to 0% and it's just fully transparent? If there's no obvious setting changes we might have to ask Microsoft support.

Reply:
> I would check a couple of things, like what do each of the property calls return if you have it enabled on your console vs disabled on your console.

when it is disabled then all properties are Default (background_color is ClosedCaptionColor::Default, background_opacity is ClosedCaptionOpacity::Default, character_edge_style is ClosedCaptionEdgeEffect::Default, etc)

when it is enabled (default) then only character_edge_style is ClosedCaptionEdgeEffect::Default and font_family is ClosedCaptionStyle::Default, rest of properties have some value (background_color is ClosedCaptionColor::Black, background_opacity is ClosedCaptionOpacity::OneHundredPercent, font_color is ClosedCaptionEdgeEffect::White, etc. 

when we choose "custom" option, we can't set "Default" to most of the properties (only to character_edge_style and font_family),


so it seems that we can simply check if background_color is different than Default if yes then we can `is_enabled = true` otherwise `is_enabled = false`.

of course I am not sure that all xbox consoles works in the same way.

The second option is omitting is_enabled flag and set: `supports_is_enabled = false;`

Reply:
I choose option 1: `is_enabled =
      ClosedCaptionProperties::BackgroundColor != ClosedCaptionColor::Default;`
      
     
Let me know if it is ok for you.

Reply:
I think that should be fine!